modules/extensions-soap-support/src/main/java/org/mule/runtime/module/extension/soap/internal/loader/InvokeOperationDeclarer.java

```diff
@@ -0,0 +1,107 @@
+import org.mule.metadata.api.ClassTypeLoader;
+import org.mule.metadata.api.annotation.TypeIdAnnotation;
+import org.mule.metadata.api.builder.BaseTypeBuilder;
+import org.mule.metadata.api.builder.UnionTypeBuilder;
+import org.mule.metadata.api.model.BinaryType;
+import org.mule.metadata.api.model.ObjectType;
+import org.mule.metadata.api.model.StringType;
+import org.mule.runtime.api.meta.model.declaration.fluent.ExtensionDeclarer;
+import org.mule.runtime.api.meta.model.declaration.fluent.OperationDeclarer;
+import org.mule.runtime.api.meta.model.declaration.fluent.ParameterGroupDeclarer;
+import org.mule.runtime.api.meta.model.display.LayoutModel;
+import org.mule.runtime.extension.api.soap.SoapAttachment;
+import org.mule.runtime.module.extension.internal.loader.java.property.ConnectivityModelProperty;
+import org.mule.runtime.module.extension.internal.loader.java.property.OperationExecutorModelProperty;
+import org.mule.runtime.module.extension.soap.internal.runtime.connection.ForwardingSoapClient;
+import org.mule.runtime.module.extension.soap.internal.runtime.operation.SoapOperationExecutorFactory;
+import org.mule.services.soap.api.message.SoapMultipartPayload;
+
+import java.io.InputStream;
+import java.util.Map;
+
+/**
+ * Declares the invoke operation for a given Soap Extension {@link ExtensionDeclarer}.
+ *
+ * @since 4.0
+ */
+public class InvokeOperationDeclarer {
+
+  static final String OPERATION_NAME = "invoke";
+  static final String OPERATION_DESCRIPTION = "invokes Web Service operations";
+
+  public static final String SERVICE_PARAM = "service";
+  public static final String OPERATION_PARAM = "operation";
+  public static final String HEADERS_PARAM = "headers";
+  public static final String REQUEST_PARAM = "request";
+  public static final String ATTACHMENTS_PARAM = "attachments";
+  public static final String TRANSPORT_HEADERS_PARAM = "transportHeaders";
+
+  private static final BaseTypeBuilder TYPE_BUILDER = BaseTypeBuilder.create(JAVA);
+
+  /**
+   * Declares the invoke operation.
+   *
+   * @param declarer the soap extension declarer
+   * @param loader   a {@link ClassTypeLoader} to load some parameters types.
+   */
+  public OperationDeclarer declare(ExtensionDeclarer declarer, ClassTypeLoader loader) {
+
+    OperationDeclarer operation = declarer.withOperation(OPERATION_NAME)
+      .describedAs(OPERATION_DESCRIPTION)
+      .withModelProperty(new OperationExecutorModelProperty(new SoapOperationExecutorFactory()));
+
+    UnionTypeBuilder output = TYPE_BUILDER.unionType();
+    output.id(Object.class.getName());
+    output.of().stringType();
+    output.of(loader.load(SoapMultipartPayload.class));
+    operation.withOutput().ofType(output.build());
+    operation.withOutputAttributes().ofType(TYPE_BUILDER.nullType().build());
+    operation.requiresConnection(true).withModelProperty(new ConnectivityModelProperty(ForwardingSoapClient.class));
+    declareParameters(operation, loader);
+
+    return operation;
```

Comment:
what about the `MetadataResolverFactoryModelProperty`? Keep in mind current metadata enrichers work only for annotation-based extensions